compiler/rustc_codegen_llvm/src/builder.rs

```diff
@@ -501,7 +501,7 @@ impl<'a, 'll, 'tcx> BuilderMethods<'a, 'tcx> for Builder<'a, 'll, 'tcx> {
             layout: TyAndLayout<'tcx>,
             offset: Size,
         ) {
-            if !scalar.is_always_valid(bx) {
+            if !scalar.is_uninit_valid() {
```

Comment:
I might suggest making the load metadata change in a separate PR--now that mem::uninit is defanged, I suspect loading from uninit heap memory (e.g. when calling the allocator directly) is more common than passing or returning uninit scalar values

Reply:
This is at least worth a perf run

Reply:
Looks like the perf benefits in rustc come from putting it on loads as well, so I'll leave it in for now.